### ISSUE TITLE
during LoRA finetuning, the training loss is NaN

### ISSUE THREAD

**Issue by jiaming-zhou** [openvla/openvla#49]
dear authors,

Thanks so much for fully releasing the OpenVLA. I am trying to fine-tune the openVLA via LoRA on berkeley_autolab_ur5/bridge_orig/ucsd_pick_and_place_dataset_converted_externally_to_rlds datasets. BUT I find the action accuracy is always 0, for all three datasets. Then I find the train_loss during my fine-tuning is always NaN. This could be the reason.

I have checked the data input, and I think they have no problem.
For the model prediction, I find 

```python
action_preds = action_logits.argmax(dim=2)
```

always predict the 0-idx token;

I obtain this info by inserting the following prints in finetune.py:
```python
            if distributed_state.is_main_process:
                print("action_logits: ", action_logits.size())
                print("action_preds: ", action_preds.size(), action_preds[0])
                print("action_gt: ", action_gt.size(), action_gt[0])
                print("mask: ", mask.size(), mask[0])
                print("correct_preds: ", correct_preds)
                print("action_accuracy: ", action_accuracy)
                print("train_loss: ", smoothened_loss)
                print("l1_loss: ", smoothened_l1_loss)
```

and the output is:
```python
action_logits:  torch.Size([1, 40, 32064])
action_preds:  torch.Size([1, 40]) tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], device='cuda:0')
action_gt:  torch.Size([1, 40]) tensor([ -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100, 31872, 31872, 31872, 31884, 31872, 31869, 31744,     2],
       device='cuda:0')
mask:  torch.Size([1, 40]) tensor([False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False,  True,  True,  True,  True,  True,  True,  True, False],
       device='cuda:0')
correct_preds:  tensor([[False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False]],
       device='cuda:0')
action_accuracy:  tensor(0., device='cuda:0')
train_loss:  nan
```

The below is the log during fine-tuning:
```python
root@ai-precog-machine9:/home/jiamingz/projects/openvla# torchrun --standalone --nnodes 1 --nproc_per_node 2 vla-scripts/finetune.py
[2024-07-26 09:00:48,903] torch.distributed.run: [WARNING] 
[2024-07-26 09:00:48,903] torch.distributed.run: [WARNING] *****************************************
[2024-07-26 09:00:48,903] torch.distributed.run: [WARNING] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
[2024-07-26 09:00:48,903] torch.distributed.run: [WARNING] *****************************************
2024-07-26 09:00:50.724369: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-07-26 09:00:50.724369: I tensorflow/core/util/port.cc:113] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-07-26 09:00:50.755711: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-07-26 09:00:50.755711: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-07-26 09:00:50.755738: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-07-26 09:00:50.755742: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-07-26 09:00:50.756664: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-07-26 09:00:50.756665: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-07-26 09:00:50.761815: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-07-26 09:00:50.761817: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-07-26 09:00:51.434507: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-07-26 09:00:51.435073: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Fine-tuning OpenVLA Model `openvla/openvla-7b` on `berkeley_autolab_ur5`
Fine-tuning OpenVLA Model `openvla/openvla-7b` on `berkeley_autolab_ur5`
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00,  5.55it/s]
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00,  5.43it/s]
trainable params: 110,828,288 || all params: 7,652,065,472 || trainable%: 1.4483
trainable params: 110,828,288 || all params: 7,652,065,472 || trainable%: 1.4483
2024-07-26 09:01:11.941738: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization
2024-07-26 09:01:12.073699: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization
07/26 [09:01:12] INFO     | >> [*] Computing dataset statistics. This may take a bit, but should only need to happen once.
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1000/1000 [02:22<00:00,  7.00it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1000/1000 [02:22<00:00,  7.00it/s]
2024-07-26 09:03:35.973908: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization
2024-07-26 09:03:35.980789: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization

######################################################################################
# Loading the following 1 datasets (incl. sampling weight):                         #
# berkeley_autolab_ur5: ====================================================1.000000 #
######################################################################################

07/26 [09:03:36] INFO     | >> [*] Threads per Dataset: [1]                                                                                           dataset.py:531
                 INFO     | >> [*] Reads per Dataset: [1]                                                                                             dataset.py:532
                 INFO     | >> [*] Constructing datasets...                                                                                           dataset.py:535

######################################################################################
# Loading the following 1 datasets (incl. sampling weight):                         #
# berkeley_autolab_ur5: ====================================================1.000000 #
######################################################################################

2024-07-26 09:03:36.700706: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization
2024-07-26 09:03:36.707621: I tensorflow/core/grappler/optimizers/data/replicate_on_split.cc:32] Running replicate on split optimization
07/26 [09:03:37] INFO     | >> [*] Applying frame transforms on dataset...                                                                            dataset.py:575
07/26 [09:03:38] INFO     | >> [*] Saved dataset statistics file at path                                                                           data_utils.py:291
                          runs/openvla-7b+berkeley_autolab_ur5+b4+lr-0.0005+lora-r32+dropout-0.0_test/dataset_statistics.json                                       
  0%|                                                                                                                                    | 0/200000 [00:00<?, ?it/s]WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
W0000 00:00:1721984618.588573  598457 op_level_cost_estimator.cc:699] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: 1 } dim { size: 224 } dim { size: 224 } dim { size: -7 } } } inputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -2 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "106" frequency: 2400 num_cores: 32 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 49152 l2_cache_size: 1310720 l3_cache_size: 25165824 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: -8 } dim { size: -9 } dim { size: -7 } } }
W0000 00:00:1721984618.588991  598457 op_level_cost_estimator.cc:699] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: 1 } dim { size: 224 } dim { size: 224 } dim { size: -6 } } } inputs { dtype: DT_FLOAT shape { dim { size: -3 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -3 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "106" frequency: 2400 num_cores: 32 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 49152 l2_cache_size: 1310720 l3_cache_size: 25165824 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -3 } dim { size: -10 } dim { size: -11 } dim { size: -6 } } }
wandb: Currently logged in as: jiamingzhou2472. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.17.5
wandb: Run data is saved locally in /home/jiamingz/projects/openvla/wandb/run-20240726_090339-90ybz7bj
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run ft+openvla-7b+berkeley_autolab_ur5+b4+lr-0.0005+lora-r32+dropout-0.0_test
wandb: ⭐️ View project at https://wandb.ai/jiamingzhou2472/openvla
wandb: 🚀 View run at https://wandb.ai/jiamingzhou2472/openvla/runs/90ybz7bj
  0%|                                                                                                                                    | 0/200000 [00:00<?, ?it/s]WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
W0000 00:00:1721984628.016364  598456 op_level_cost_estimator.cc:699] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: 1 } dim { size: 224 } dim { size: 224 } dim { size: -7 } } } inputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -2 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "106" frequency: 2400 num_cores: 32 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 49152 l2_cache_size: 1310720 l3_cache_size: 25165824 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: -8 } dim { size: -9 } dim { size: -7 } } }
W0000 00:00:1721984628.016811  598456 op_level_cost_estimator.cc:699] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: 1 } dim { size: 224 } dim { size: 224 } dim { size: -6 } } } inputs { dtype: DT_FLOAT shape { dim { size: -3 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -3 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "106" frequency: 2400 num_cores: 32 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 49152 l2_cache_size: 1310720 l3_cache_size: 25165824 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -3 } dim { size: -10 } dim { size: -11 } dim { size: -6 } } }
action_logits:  torch.Size([1, 40, 32064])
action_preds:  torch.Size([1, 40]) tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], device='cuda:0')
action_gt:  torch.Size([1, 40]) tensor([ -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100, 31872, 31872, 31872, 31884, 31872, 31869, 31744,     2],
       device='cuda:0')
mask:  torch.Size([1, 40]) tensor([False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False,  True,  True,  True,  True,  True,  True,  True, False],
       device='cuda:0')
correct_preds:  tensor([[False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False]],
       device='cuda:0')
action_accuracy:  tensor(0., device='cuda:0')
train_loss:  nan

l1_loss:  0.8638655462184873
action_logits:  torch.Size([1, 40, 32064])
action_preds:  torch.Size([1, 40]) tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], device='cuda:0')
action_gt:  torch.Size([1, 40]) tensor([ -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,  -100,
         -100,  -100, 31872, 31872, 31872, 31884, 31872, 31869, 31744,     2],
       device='cuda:0')
mask:  torch.Size([1, 40]) tensor([False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, False, False, False, False, False,
        False, False,  True,  True,  True,  True,  True,  True,  True, False],
       device='cuda:0')
correct_preds:  tensor([[False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False]],
       device='cuda:0')
action_accuracy:  tensor(0., device='cuda:0')
train_loss:  nan
...
...
```

can you find the cause of this problem? many thanks!

**Comment by jiaming-zhou** [comment]
I found that the problem is caused by loading wrong pre-training weights. After using the right weights, everything goes well. I will close the issue.
